src/react/client.tsx

```diff
@@ -226,19 +249,17 @@ export function AuthProvider({
           const refreshToken =
             (await storageGet(REFRESH_TOKEN_STORAGE_KEY)) ?? null;
           if (refreshToken !== null) {
+            setIsRefreshingToken(true);
```

Comment:
I moved this up slightly because it feels like we probably shouldn't let you navigate away when we've cleared out your old token but before we've fetched the new one.

If `verifyCodeAndSetToken` fails for some reason, we'll treat this as if you've been logged out, but let you close the tab.

Reply:
Know what the goal of not allowing navigation away during this time was? Could there be queued mutations that wouldn't be covered by the existing warning?

Reply:
My best guess is that we didn't want the user closing the tab at an inopportune moment to result in them being logged out when they come back?